Fix type annotations in pandas.core.base

Part of #25882 

pandas.core.base has the following mypy errors:
```
pandas/core/base.py:840: error: "IndexOpsMixin" has no attribute "_values"; maybe "_map_values"?
pandas/core/base.py:963: error: "IndexOpsMixin" has no attribute "values"
```

Pull request to follow

--- a/mypy.ini
+++ b/mypy.ini
@@ -8,9 +8,6 @@ ignore_errors=True
 [mypy-pandas.core.api]
 ignore_errors=True
 
-[mypy-pandas.core.base]
-ignore_errors=True
-
 [mypy-pandas.core.computation.expr]
 ignore_errors=True
 

--- a/pandas/core/base.py
+++ b/pandas/core/base.py
@@ -837,7 +837,10 @@ class IndexOpsMixin:
         [a, b, a]
         Categories (2, object): [a, b]
         """
-        result = self._values
+        # As a mixin, we depend on the mixing class having _values.
+        # Special mixin syntax may be developed in the future:
+        # https://github.com/python/typing/issues/246
+        result = self._values  # type: ignore
 
         if is_datetime64_ns_dtype(result.dtype):
             from pandas.arrays import DatetimeArray
@@ -960,7 +963,10 @@ class IndexOpsMixin:
         """
         if is_extension_array_dtype(self):
             return self.array._ndarray_values
-        return self.values
+        # As a mixin, we depend on the mixing class having values.
+        # Special mixin syntax may be developed in the future:
+        # https://github.com/python/typing/issues/246
+        return self.values  # type: ignore
 
     @property
     def empty(self):

